src/App.jsx

```diff
@@ -1,24 +1,103 @@
+    const handleColorChange = (e) => {
+        const selectedColor = e.target.value;
+        console.log("Color changed:", selectedColor);
+        setColor(selectedColor);
+    };
 
-      {/* Conditionally render based on wether the user is inputting an URL to generate a QR Code or the user wnats to downaload the generated QR Code from the url input */}
-      {/* {yourReactiveVariableThatTogglesTheDownloadQrCcodeOrInputField ? () : ()} */}
-    </div>
-  );
+    return (
+        <div className="app">
+            <div className="app-container">
+                <div className="header">
+                    <img className="logo" src={logo} alt="" />
+                    <h1 className="header-h1">QR CODE GENERATOR</h1>
+                </div>
+                <div className="generator">
+                    {isLoading ? (
+                        <div className="loading-container">
+                            <LoadingAnimation />
+                        </div>
+                    ) : showInput ? (
+                        <>
+                            <QrAnimation />
+                            <h3>
+                                Want a QR code? You've come to the right place!
+                                Just paste in your url here, and click generate!
+                                You'll be able to download it once the QR code
+                                is generated.{" "}
+                            </h3>
+                            <input
+                                className="url-input"
+                                type="text"
+                                placeholder="e.g. https://google.com"
+                                value={url}
+                                onChange={(e) => setUrl(e.target.value)}
+                            />
+                            <div className="color-select-container">
+                                <span
+                                    className="color-preview"
+                                    style={{ backgroundColor: color }}
+                                />
+
+                                <select
+                                    name="colorSelector"
+                                    className="color-selector"
+                                    value={color}
+                                    onChange={handleColorChange}
+                                >
+                                    <option disabled value="">
+                                        Color
+                                    </option>
+                                    <option value="#020617">Black</option>
+                                    <option value="#86198f">Pink</option>
+                                    <option value="#3730a3">Indigo</option>
+                                    <option value="#4d7c0f">Green</option>
+                                    <option value="#b45309">Orange</option>
+                                </select>
```

Comment:
Really nice color selector! I like that the user gets a few options to select from, it looks really good, and it works. great job!